src/Microsoft.AspNetCore.OData/Batch/ODataBatchStream.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace Microsoft.AspNet.OData.Batch
+{
+    // Normally, the stream is closed when the ODataMessageWriter is disposed.
+    // For responses within a batch, we need to read from the stream after the response is
+    // written in order to write it to the batch response stream. So we need to ignore the Close()
+    // and provide an alternate method to release the stream after writing its content to the batch response.
+    internal class BatchStream : MemoryStream
+    {
+        /// <summary>
+        /// Release the batch stream and underlying resources
+        /// </summary>
+        internal void Release()
```

Comment:
>internal void Release() [](start = 8, length = 23)

shall we add a protect mechanism? 
how about to call twice "Release()"? 

Maybe test IsClosed() before call base.Close()?